context.go

```diff
@@ -0,0 +1,162 @@
+	Traits    map[string]interface{} `json:"traits,omitempty"`
+
+	// This map is used to allow extensions to the context specifications that
+	// may not be documented or could be introduced in the future.
+	// The fields of this map are inlined in the serialized context object,
+	// there is no actual "extra" field in the JSON representation.
+	Extra map[string]interface{} `json:"-"`
+}
+
+// This type provides the representation of the `context.app` object as defined
+// in https://segment.com/docs/spec/common/#context
+type AppInfo struct {
+	Name    string `json:"name,omitempty"`
+	Version string `json:"version,omitempty"`
+	Build   string `json:"build,omitempty"`
+}
+
+// This type provides the representation of the `context.campaign` object as
+// defined in https://segment.com/docs/spec/common/#context
+type CampaignInfo struct {
+	Name    string `json:"name,omitempty"`
+	Source  string `json:"source,omitempty"`
+	Medium  string `json:"medium,omitempty"`
+	Term    string `json:"term,omitempty"`
+	Content string `json:"content,omitempty"`
+}
+
+// This type provides the representation of the `context.device` object as
+// defined in https://segment.com/docs/spec/common/#context
+type DeviceInfo struct {
+	Id           string `json:"id,omitempty"`
+	Manufacturer string `json:"manufacturer,omitempty"`
+	Model        string `json:"model,omitempty"`
+	Name         string `json:"name,omitempty"`
+	Type         string `json:"type,omitempty"`
+	Version      string `json:"version,omitempty"`
+}
+
+// This type provides the representation of the `context.library` object as
+// defined in https://segment.com/docs/spec/common/#context
+type LibraryInfo struct {
+	Name    string `json:"name,omitempty"`
+	Version string `json:"version,omitempty"`
+}
+
+// This type provides the representation of the `context.location` object as
+// defined in https://segment.com/docs/spec/common/#context
+type LocationInfo struct {
+	City      string  `json:"city,omitempty"`
+	Country   string  `json:"country,omitempty"`
+	Region    string  `json:"region,omitempty"`
+	Latitude  float64 `json:"latitude,omitempty"`
+	Longitude float64 `json:"longitude,omitempty"`
+	Speed     float64 `json:"speed,omitempty"`
+}
+
+// This type provides the representation of the `context.network` object as
+// defined in https://segment.com/docs/spec/common/#context
+type NetworkInfo struct {
+	Bluetooth bool   `json:"bluetooth,omitempty"`
+	Cellular  bool   `json:"cellular,omitempty"`
+	WIFI      bool   `json:"wifi,omitempty"`
+	Carrier   string `json:"carrier,omitempty"`
+}
+
+// This type provides the representation of the `context.os` object as defined
+// in https://segment.com/docs/spec/common/#context
+type OSInfo struct {
+	Name    string `json:"name,omitempty"`
+	Version string `json:"version,omitempty"`
+}
+
+// This type provides the representation of the `context.page` object as
+// defined in https://segment.com/docs/spec/common/#context
+type PageInfo struct {
```

Comment:
Here I've suffixed the context sub-types with `-Info` because there's already a type named `Page` (for the page message).
I noticed that the Java library was naming the messages with a `-Message` suffix (like `PageMessage`), if we were following this naming convention we could get rid of the `-Info` suffix. However I feel like `Track`, `Page`, `Identify`, etc... are used way more often than the context sub-types so I feel better about keeping the shorter names for what's most common.
Please let me know if you have a different point of view.